File: Java/Pesawat.java
class Pesawat extends Kendaraan {

    private String maskapaiPemilik;
    private int ketinggianMaksimalKaki;
    private String rutePenerbangan;
    private int jumlahMesin;

    public Pesawat() {
        super();
    }

    // konstruktor dengan parameter
    public Pesawat(String jenisBahanBakar, int kapasitasPenumpang, int tahunProduksi, String statusMesin,
                    String maskapaiPemilik, int ketinggianMaksimalKaki, String rutePenerbangan, int jumlahMesin) {
        super(jenisBahanBakar, kapasitasPenumpang, tahunProduksi, statusMesin);
        this.maskapaiPemilik = maskapaiPemilik;
        this.ketinggianMaksimalKaki = ketinggianMaksimalKaki;
        this.rutePenerbangan = rutePenerbangan;
        this.jumlahMesin = jumlahMesin;
    }

    // getter setter
    public String getMaskapaiPemilik() {
        return maskapaiPemilik;
    }

    public void setMaskapaiPemilik(String maskapaiPemilik) {
        this.maskapaiPemilik = maskapaiPemilik;
    }

    public int getKetinggianMaksimalKaki() {
        return ketinggianMaksimalKaki;
    }

    public void setKetinggianMaksimalKaki(int ketinggianMaksimalKaki) {
        this.ketinggianMaksimalKaki = ketinggianMaksimalKaki;
    }

    public String getRutePenerbangan() {
        return rutePenerbangan;
    }

    public void setRutePenerbangan(String rutePenerbangan) {
        this.rutePenerbangan = rutePenerbangan;
    }

    public int getJumlahMesin() {
        return jumlahMesin;
    }

    public void setJumlahMesin(int jumlahMesin) {
        this.jumlahMesin = jumlahMesin;
    }
}

File: Cpp/Pesawat.cpp
#include <iostream>
#include "Kendaraan.cpp"
using namespace std;

class Pesawat : public Kendaraan {

    private:
        string maskapaiPemilik;
        int ketinggianMaksimalKaki;
        string rutePenerbangan;
        int jumlahMesin;

    public:
        Pesawat() {
        }

        // konstruktor dengan parameter
        Pesawat(string jenisBahanBakar, int kapasitasPenumpang, int tahunProduksi, string statusMesin,
                string maskapaiPemilik, int ketinggianMaksimalKaki, string rutePenerbangan, int jumlahMesin)
            : Kendaraan(jenisBahanBakar, kapasitasPenumpang, tahunProduksi, statusMesin) {
            this->maskapaiPemilik = maskapaiPemilik;
            this->ketinggianMaksimalKaki = ketinggianMaksimalKaki;
            this->rutePenerbangan = rutePenerbangan;
            this->jumlahMesin = jumlahMesin;
        }

        // getter setter
        string getMaskapaiPemilik() {
            return maskapaiPemilik;
        }

        void setMaskapaiPemilik(string maskapaiPemilik) {
            this->maskapaiPemilik = maskapaiPemilik;
        }

        int getKetinggianMaksimalKaki() {
            return ketinggianMaksimalKaki;
        }

        void setKetinggianMaksimalKaki(int ketinggianMaksimalKaki) {
            this->ketinggianMaksimalKaki = ketinggianMaksimalKaki;
        }

        string getRutePenerbangan() {
            return rutePenerbangan;
        }

        void setRutePenerbangan(string rutePenerbangan) {
            this->rutePenerbangan = rutePenerbangan;
        }

        int getJumlahMesin() {
            return jumlahMesin;
        }

        void setJumlahMesin(int jumlahMesin) {
            this->jumlahMesin = jumlahMesin;
        }
};